### PR TITLE
Cleanup cli

### PR DESCRIPTION
**Pull request by lambdaFun94**
streamlining our complex CLI. 
1. `npm run dev` command is all batteries included now: will pull local data
2. `build:serve` can be used for the final preview before pushing
Only time that `run dev` needs to be rerun is if stream configuration is changed. 
cc @Mxs2019 @mkilpatrick

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,8 +6,11 @@
   "license": "BSD-3-Clause",
   "type": "module",
   "scripts": {
-    "dev": "pages dev",
-    "build": "vite build",
+    "dev": "yext pages build && yext pages generate-test-data && pages dev",
```
Comment (mkilpatrick) [inline package.json:9]: For dev we don't need the artifact-output, so change `yext pages build` to `vite build` or `npm run build:local`. I'd prefer the latter for the same reason as in my below comment.

File: package.json [hunk 2]
```diff
@@ -6,8 +6,11 @@
...
+    "build:local": "vite build",
+    "build": "yext pages build && yext pages generate-test-data && yext pages render",
+    "build:serve": "yext pages build && yext pages generate-test-data && yext pages render && yext pages serve",
```
Comment (mkilpatrick) [inline package.json:12]: I'd just have this call `npm run build && npm run serve` to keep it simpler. Also, if you change one command in the future you don't need to change in both places.

Comment (lambdaFun94) [reply package.json]: @mkilpatrick done.

**Review by mkilpatrick — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -22,26 +24,36 @@ cd pages-starter-react-locations
...
+While _developing locally_, run the following command:
+
+```
+npm run dev
+```
+
+  This command will start a Vite-powered dev server that will enable hot-reloading. Additionally, the command will generate a `localData` directory that contains a subset of your Knowledge Graph data. This command is automatically in "dynamic" mode, which means it will pull data updates automatically from your Knowledge graph, so real-time data changes in your Yext account will be reflected in your local dev site.
 
-`npm run dev` - runs your code against a local dev server using Vite
+  NOTE: Whenever you make changes to your stream definitions, you must re-run `npm run dev` for the system to updates the `features.json` and the required entities to power your site.
```
Comment (mkilpatrick) [inline README.md:37]: "for the system to update" - no "s"

File: README.md [hunk 4]
```diff
@@ -22,26 +24,36 @@ cd pages-starter-react-locations
...
-- All stream documents come from the `localData` folder
-- You can visit either of these urls out of the box
-  - http://localhost:3000/index/123
-  - http://localhost:3000/static
+_Before committing_ your code, we recommend running the following command:
+```
+npm run build:serve
+``` 
+
+  This command will generate a production build of your site, so you can ensure there are no build errors or unexpected behavior. This build step replicates the production build environment used in the Yext system, and serves your data at `localhost:8000`.
```
Comment (mkilpatrick) [inline README.md:44]: Remove indent for these paragraphs

File: README.md [hunk 5]
```diff
@@ -22,26 +24,36 @@ cd pages-starter-react-locations
...
+
+  In practice, development builds (via `npm run dev`) and production builds compile and bundle assets differently. For local development, ES Modules are loaded directly by the browser, allowing fast iteration during local development and also allows for hot module replacement (HMR). Other things like CSS are also loaded directly by the browser, including linking to sourcemaps. During a production build all of the different files are compiled (via ESBuild for jsx/tsx) and minified, creating assets as small as possible so that the final html files load quickly when served to a user.
```
Comment (mkilpatrick) [inline README.md:46]: You might also want to mention "tree-shaken" in the compiled/minified sentence

File: README.md [hunk 6]
```diff
@@ -22,26 +24,36 @@ cd pages-starter-react-locations
...
+### Other Useful commands
 
-`yext pages generate-test-data` - pull an example set of `localData` from your account
+`yext init` - Authenticates the Yext CLI with your Yext account
 
-`yext pages build` - Runs a production build against your `localData`
+`yext pages generate-test-data` - pull an example set of `localData` from your account. This command is packaged within `npm run dev'. 
 
-`yext pages serve` - Runs a local server against your production-built files
+`yext pages build` - Runs a production build against your `localData`: part one of `npm run build:serve`
```
Comment (mkilpatrick) [inline README.md:54]: Is this supposed to be `npm run build`?

File: README.md [hunk 7]
```diff
@@ -76,7 +89,9 @@ NOTE: You normally wouldn't want to check in the localData folder as it's only u
 
 Contains a single `ci.json` file. This file defines how the Yext CI system will build your project. It is not used during local dev. However, it is used when running a local production build (i.e. `yext pages build`).
 
-NOTE: A `features.json` file will automatically be generated during CI build for you based on the `config`s defined in your templates. One has been checked in to this repo so that `yext pages generate-test-data` works out of the box (assuming you've `yext init`'ed with your Yext account). If this file doesn't exist then `yext pages build` will implicitly generate a new one when it calls `npm run directbuild` (defined in `sites-config/ci.json`).
+NOTE: A `features.json` file will automatically be generated during CI build for you based on the template configs defined in your templates. If this file doesn't exist then `yext pages build` will implicitly generate a new one when it calls `npm run build` (defined in `sites-config/ci.json`). In the recommended devleopment flow with `npm run dev`, the `features.json` will be automatically generated.
```
Comment (mkilpatrick) [inline README.md:92]: Update the `npm build build` part
> If this file doesn't exist then `yext pages build` will implicitly generate a new one when it calls `npm run build` (defined in `sites-config/ci.json`).

to `npm run build:local`

File: README.md (hunk 5 above)
Comment (lambdaFun94) [reply README.md:46]: @mkilpatrick can you give more detail about what you'd like me to mention? I just want to make sure I'm accurate in what I"m saying.

Comment (mkilpatrick) [reply README.md]: tree-shaking happens during bundling, which is the process of removing any dependencies that aren't actually used. I was just saying to mention it here. Most devs should know what it means.

**Comment by lambdaFun94** [comment]
@mkilpatrick all comments resolved

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**